service/src/test/java/com/epam/rft/atsy/service/impl/CandidateServiceImplTest.java

```diff
@@ -0,0 +1,212 @@
+
+    private static final Long ID = 1L;
+    private static final String NAME = "John Doe";
+    private static final String EMAIL = "john@doe.com";
+    private static final String PHONE = "123456";
+    private static final String REFERER = "Jane Doe";
+    private static final Short LANGUAGE_SKILL = 5;
+    private static final String DESCRIPTION = "Simply John Doe.";
+
+    private static final String SORT_FIELD = "name";
+
+    private static final CandidateEntity NOT_FOUND_CANDIDATE_ENTITY = null;
+    private static final CandidateDTO NOT_FOUND_CANDIDATE_DTO = null;
+
+    private CandidateEntity dummyCandidateEntity;
+
+    private CandidateDTO dummyCandidateDto;
+
+    private FilterRequest ascendingFilterRequest;
+
+    private FilterRequest descendingFilterRequest;
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Before
+    public void setUp() {
+        dummyCandidateEntity = CandidateEntity.builder().id(ID).name(NAME).email(EMAIL).phone(PHONE)
+                                              .referer(REFERER).languageSkill(LANGUAGE_SKILL)
+                                              .description(DESCRIPTION).build();
+
+        dummyCandidateDto = CandidateDTO.builder().id(ID).name(NAME).email(EMAIL).phone(PHONE)
+                                        .referer(REFERER).languageSkill(LANGUAGE_SKILL)
+                                        .description(DESCRIPTION).build();
+
+        ascendingFilterRequest = FilterRequest.builder().fieldName(SORT_FIELD).order(SortingRequest.Order.ASC)
+                                              .searchOptions(new SearchOptions(NAME, EMAIL, PHONE)).build();
+
+        descendingFilterRequest = FilterRequest.builder().fieldName(SORT_FIELD).order(SortingRequest.Order.DESC)
+                                              .searchOptions(new SearchOptions(NAME, EMAIL, PHONE)).build();
+    }
+
+    @Test
+    public void getCandidateShouldReturnNullWhenIdNotFound() {
+        // Given
+        given(candidateRepository.findOne(ID)).willReturn(NOT_FOUND_CANDIDATE_ENTITY);
+        given(modelMapper.map(NOT_FOUND_CANDIDATE_ENTITY, CandidateDTO.class)).willReturn(NOT_FOUND_CANDIDATE_DTO);
+
+        // When
+        CandidateDTO candidate = candidateService.getCandidate(ID);
+
+        // Then
+        assertThat(candidate, nullValue());
+    }
+
+    @Test
+    public void getCandidateShouldReturnProperDtoWhenIdExists() {
+        // Given
+        given(candidateRepository.findOne(ID)).willReturn(dummyCandidateEntity);
+        given(modelMapper.map(dummyCandidateEntity, CandidateDTO.class)).willReturn(dummyCandidateDto);
+
+        // When
+        CandidateDTO candidate = candidateService.getCandidate(ID);
+
+        // Then
+        assertThat(candidate, equalTo(dummyCandidateDto));
+    }
+
+    @Test
+    public void getAllCandidateShouldCallRepositoryWithAscendingSortWhenPassingAscendingFilterRequest() {
```

Comment:
Also define the expected behavior for invalid input.
eg: filterRequest is null, or not properly initialized